### ISSUE TITLE
Is there any way to send the agent group with the whole log the agent sends?

### ISSUE THREAD

**Issue by dave36** [wazuh/wazuh#5243]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12.3 | Wazuh component | Agent | Packages | Windows 10 |

### Question
Is there any way to send the agent group with the whole log the agent sends?

### Description
We have multiple agents installed on two clients and we want to distinguish each client with his devices. We are able to do it with a huge if condition in the logstash configuration and comparing the name of each device to assign the index it fits. However, if a client has too many devices, this is not scalable.

So we tried using agent groups to put together the devices of each client. If we look into the Wazuh App on Kibana, they show up perfectly, but in the Discover tab, in the content of a log, we cannot see the field group. Therefore, we are not able to discern between a client from that field because it is not there. It would be perfect if the agent could send the group which belongs. Is there any kind of workaround to achieve this?

Thanks in advance for your incredible job,
David

**Comment by pereyra-m** [comment]
Hello David.

Thank you for your message, let me try to clarify the situation:

Have you already used the [Agent labels](https://documentation.wazuh.com/3.12/user-manual/reference/ossec-conf/labels.html) and they are not working ?

Because if you didn't, you can add a **labels** section in the **ossec.conf** file in the agent.

```
<labels>
  <label key="client-ID">1234</label>
  <label key="agent_group">group-1</label>
  <label key="network.ip">172.17.0.0</label>
  <label key="network.mac">02:42:ac:11:00:02</label>
</labels>
```
This info is sent to the manager in every alert reported by the agent.

You can add to your question an example of the alerts log file and a picture of the Kibana app, so we can understand each other better.

I'll be waiting your response.
Regards.

**Comment by dave36** [comment]
Hello,

Yes, we have already tried the Agent labels but we still could not see it on the Kibana tab "Discover". In the ossec.conf we added:

```
<labels>
  <label key="agent.group">group1</label>
</labels>
```

And in the alerts.log on the Wazuh Manager we could see it, but before the event and not in the same level as other fields:
```
** Alert 1592384987.324368663: - windows, windows_security,authentication_success,pci_dss_10.2.5,gpg13_7.1,gpg13_7.2
,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,
2020 Jun 17 11:09:47 (PORTXXXX) any->EventChannel
agent.group: group1
Rule: 60106 (level 3) -> 'Windows Logon Success'
{"win":{"system":{"providerName":"Microsoft-Windows-Security-Auditing","providerGuid":"{54849625-5478-4994-a5ba-3e3b
0328c30d}","eventID":"4624","version":"2","level":"0","task":"12544","opcode":"0","keywords":"0x8020000000000000","s
ystemTime":"2020-06-17T09:09:45.036608800Z","eventRecordID":"104622","processID":"840","threadID":"27228","channel":
"Security", ....
```

Alternately, in the alerts.json:
```
{
    "timestamp": "2020-06-17T11:23:37.024+0200",
    "rule": {
        "level": 3,
        "description": "Windows User Logoff",
        "id": "60137",
        "firedtimes": 102,
        "mail": false,
        "groups": ["windows", " windows_security"],
        "pci_dss": ["10.2.5"],
        "gdpr": ["IV_32.2"],
        "hipaa": ["164.312.b"],
        "nist_800_53": ["AU.14", "AC.7"]
    },
    "agent": {
        "id": "001",
        "name": "PORTXXXX",
        "ip": "192.168.0.16",
        "labels": {
            "agent": {
                "group": "group1"
            }
        }
    }, ....
```

In the Discover tab of Kibana, the agent's field we receive were this but not the new label added before:
![image](https://user-images.githubusercontent.com/18281060/84985161-8e31bd80-b13c-11ea-983b-59da79dafcc1.png)

I do not know why on Kibana we are not able to see that field of the label. If we could see it, the problem to distinguish would be solved.

Thanks in advance for your help,
David

**Comment by dave36** [comment]
Ok, we have already fixed it. It seems that Filebeat was changing some information before sending it. Instead of:
`"agent":{"id":"001","name":"PORTXXXX",...`

In Kibana we were seeing:
`"agent":{"id":"4e41d0a6-f7df-483d-9f90-xxxxxxxxx",.....`

And changing the config of Filebeat in the Wazuh Manager (/etc/filebeat/filebeat.yml) it worked!!. It stopped changing that information and started sending the agent's label. We added this:
```
processors:
  - convert:
      fields:
        - {from: "agent", to: "filebeat_agent"}
      ignore_missing: true
      fail_on_error: false
      mode: rename
```

It is related to [this one](https://github.com/wazuh/wazuh-documentation/issues/1862).

Thank you very much for your dedication and effort!
Cheers!
